src/System.Text.Encoding.CodePages/src/System/Text/ISO2022Encoding.cs

```diff
@@ -1498,8 +1498,7 @@ private ISO2022Modes CheckEscapeSequenceKR(byte[] bytes, int escapeCount)
                         }
 
                         // Stick it in decoder
-                        if (decoder != null)
```

Comment:
>     if (decoder != null) [](start = 19, length = 25)

I am not sure if removing this is correct. if the method GetCharsCP52936 is always called with valid decoder, then we should make it as not nullable and clean the code which check decoder==null. If GetCharsCP52936 is called with nullable decoder, we'll need to keep this check then and just move the next few lines under the condition.

Reply:
@tarekgh, just before this there's this code:
```C#
if (decoder == null || ...)
{
    ...
    break;
}
```
How could `decoder` be `null` here?

Reply:
you are right. I didn't notice the break. so we are fine then. 

---
In reply to: [337629043](https://github.com/dotnet/corefx/pull/41991#discussion_r337629043) [](ancestors = 337629043)